### ISSUE TITLE
Weird expansion bug with `cmp`, that repeats trigger at first insert point :thinking: 

### ISSUE THREAD

**Issue by bew** [L3MON4D3/LuaSnip#1019]
I'm not sure which plugin is doing this behavior exactly

Using the following snippet:
```lua
snip("do", {desc = "do ... end"}, U.myfmt {
  [[
    do
      <body>
    end
  ]],
  { body = i(0) },
})
```

![Peek 2023-09-28 00-12](https://github.com/L3MON4D3/LuaSnip/assets/9730330/ce7ec20d-3169-4ec1-a958-7e3175d1406c)

Here I try to use the snippet 4 times:
1. with no text before, direct expansion via `ls.expand()`
2. with no text before, with expansion through `cmp` (select then `<C-y>`)
3. with text before, direct expansion via `ls.expand()`
4. with text before, with expansion through `cmp` (select then `<C-y>`)

I'm using `cmp` with `cmp_luasnip` of course

**Comment by L3MON4D3** [comment]
Can't reproduce this one either :/

https://github.com/L3MON4D3/LuaSnip/assets/41961280/4089c07e-06eb-41bc-864f-938d9cf96cb8

**Comment by bew** [comment]
Yeah that's what I thought, I'll try to repro with a minimal config later, probably this weekend.

**Comment by bew** [comment]
It might be because I'm still using cmp's native menu.

Here is a reproduction script:
https://gist.github.com/bew/76f2d512ec42136f61b3802335a7b620
![Peek 2023-10-08 07-08](https://github.com/L3MON4D3/LuaSnip/assets/9730330/aa726bf8-0830-4850-9cac-d45e1bc03b3a)
